react/basic-example/src/components/TalkSession.js

```diff
@@ -0,0 +1,47 @@
+/* 
+This file acts as a provider that all children components can directly access
+the talkSession context and is useful if you may want to interact with TalkJS in different parts 
+of your application. 
+*/ 
+
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Talk from 'talkjs';
+import { TalkSessionContext } from '../contexts/TalkSessionContext';
+
+
+function TalkSession(props) {
+
+    const [me, setMe] = useState(null);
+    const [talkSession, setTalkSession] = useState(null);
+
+    useEffect(() => {
+        
+        if(!talkSession) {
+            Talk.ready.then(() => {
+                const me = new Talk.User(props.me);
+                const session = new Talk.Session({
+                    me,
+                    appId: props.appId,
+                });
+    
+                setMe(me);
+                setTalkSession(session);
+            });
+        }
+    }, [props, talkSession]);
```

Comment:
```suggestion
    }, [props.me, props.appId]);
```
1. i'd remove talkSession here, i don't think there's any use to running the effect again if it changes - after all, only the effect itself can change it
2. separating out the props ensures that it's not rerun on every render (because the project object is always fresh, but each individual prop is often the same as in the previous render).

Reply:
I think talkSession belongs here, as initially it'll be `null` and we want the effect to run when it changes from null, plus it throws an eslint warning without it. Good point about changing it to props.me though and props.appId - I'll add that.

Reply:
> I think talkSession belongs here, as initially it'll be null and we want the effect to run when it changes from null

Why? I mean, the effect is the only thing makes it change. So it'll run after the first render, `talkSession` will be null so the effect calls `setTalkSession`. This will trigger the effect to run again (because `talkSession` changed from `null` to a Talk.Session instance), and then the `if` is false so it immediately returns. That second run serves no purpose.

Reply:
Ah yeah true, although I don't like shipping something that is throwing ESlint warnings each page load / render etc. 

> React Hook useEffect has a missing dependency: 'talkSession'. Either include it or remove the dependency array.

Reply:
One solution would be to remove the `if(!session)` as well. That would recreate a new session every time `me` or `appId` change but, well, that does make some sense. 

In fact, I think it might be good form to add `return () => session?.destroy();` to the last line of useEffect. This will clean up if the TalkSession gets unmounted, and also will make changing `me` or `appId` actually work.